_posts/2022-10-12-ecies-vau.markdown

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title:  "VAU and the Brainpool; plus some Kotlin - a more lightweight approach to ECIES"
+date:   2022-10-12 11:51:12 +0200
+author: Tobias Schwerdtfeger
+categories: tech
+tags: ECIES VAU TEE Kotlin Java Crypto BouncyCastle Brainpool NIST
+---
+
+Probably you're here for an answer on how to implement the Elliptic Curve Integrated Encryption Scheme (ECIES) in
+something like Kotlin or Java for communicating with the trusted execution environment
+(TEE; or in german, we like to call this VAU) of our e-prescription server.
+Perfect! Because otherwise, I would lose your attention to three lines of code:
+
+```kotlin
+val cipher = Cipher.getInstance("ECIES", BCProvider)
+cipher.init(Cipher.ENCRYPT_MODE, publicKey)
+cipher.doFinal(content)
+```
+
+After your initial excitement, I've to tell you the truth: The code above is not sufficient for our needs 😔
+
+**BUT!** If you only want to use ECIES with some defaults (like NIST curves) using Android, I can recommend to use [Tink](https://github.com/google/tink/blob/master/docs/JAVA-HOWTO.md#hybrid-encryption).
+
+_Disclaimer: I'm not a crypto expert who can answer the ins and outs of cryptographic schemes or elliptic curves.
+I just know how to implement this stuff. And it works! Well, now, not in the beginning...
+So if you only want to survive the loveliness of using rarely used elliptic curves and not so easy to implement cipher suits, you're welcome!_
+
+## How do we got here?
```

Comment:
got -> get

Reply:
For some reason, for me it sounds always better with "got" haha